### PR TITLE
Minor renaming and fix of error handling from admin portal

### PR DESCRIPTION
**Pull request by pestanko**
Renamed errors to error and
removed Rspec prefix

Signed-off-by: Peter Stanko <pstanko@redhat.com>

### CODE REVIEW

File: lib/three_scale_api/tools.rb [hunk 1]
```diff
@@ -52,8 +52,8 @@ def self.log_factory(log_level: 'debug')
     #
     # @param [Hash] response Response from server
     def self.check_response(response)
-      if (response.is_a? Hash) && !response['error'].nil?
-        raise APIResponseError.new(response), response['error']
+      if (response.is_a? Hash) && (!response['error'].nil? || !response['errors'].nil?)
```
Comment (mikz) [inline lib/three_scale_api/tools.rb]: `!response['error'].nil?` is almost the same as `response['error']`. I'd suggest just doing `response['error']` which looks way better.

File: spec/integration/application_plan_limit_spec.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 require_relative '../shared_tests_config'
 
-RSpec.describe 'Application plan limit resource', type: :integration do
+describe 'Application plan limit resource', type: :integration do
```
Comment (mikz) [inline spec/integration/application_plan_limit_spec.rb]: This is against RSpec best practices. It is not recommended to pollute global namespace. Default rspec installation will forbid this. I'd recommend using the RSpec namespace.

Comment (pestanko) [inline spec/integration/application_plan_limit_spec.rb]: ok, I will give it back

**Comment by pestanko** [comment]
@mikz fixed.

File: lib/three_scale_api/tools.rb [hunk 3]
```diff
@@ -52,8 +52,8 @@ def self.log_factory(log_level: 'debug')
     #
     # @param [Hash] response Response from server
     def self.check_response(response)
-      if (response.is_a? Hash) && !response['error'].nil?
-        raise APIResponseError.new(response), response['error']
+      if (response.is_a? Hash) && (!response['error'] || !response['errors'])
```
Comment (mikz) [inline lib/three_scale_api/tools.rb]: This breaks the logic. Before you had `!response['error'].nil?` and now `!response['error']`. Removing `!` would keep it.

Comment (pestanko) [inline lib/three_scale_api/tools.rb]: @mikz Yes you are right, my bad, I have fixed it.

Comment (mikz) [inline lib/three_scale_api/tools.rb]: No worries. Thats what are reviews for 👍

**Pull request merged**